bftengine/src/preprocessor/RequestProcessingState.cpp

```diff
@@ -167,6 +167,34 @@ bool RequestProcessingState::isReqTimedOut() const {
   return false;
 }
 
+std::pair<std::string, concord::util::SHA3_256::Digest> RequestProcessingState::detectFailureDueToBlockID(
+    const concord::util::SHA3_256::Digest &other, uint64_t blockId) {
+  // since this scenario is rare, a new string is allocated for safety.
+  std::string modifiedResult(primaryPreProcessResult_, primaryPreProcessResultLen_);
+  ConcordAssertGT(modifiedResult.size(), sizeof(uint64_t));
+  memcpy(modifiedResult.data() + modifiedResult.size() - sizeof(uint64_t),
+         reinterpret_cast<char *>(&blockId),
+         sizeof(uint64_t));
+  auto modifiedHash = convertToArray(SHA3_256().digest(modifiedResult.c_str(), modifiedResult.size()).data());
+  if (other == modifiedHash) {
+    LOG_INFO(logger(), "Primary hash is different from quorum due to mismatch in block id");
```

Comment:
Your logs don't have any reference to the command. How are you going to look for them in the logs?